generatorreceiver/internal/topology/latency_percentiles.go

```diff
@@ -29,20 +29,26 @@ type LatencyPercentiles struct {
 }
 
 func (l *LatencyPercentiles) Sample() int64 {
+	if l == nil {
+		// This results from having a list where
+		// items are !ShouldGenerate() which leaves
+		// an empty list, the zeroP is returned.
+		return 0
+	}
 	uniform := func(timeA, timeB time.Duration) int64 {
 		min := float64(timeA.Nanoseconds())
 		max := float64(timeB.Nanoseconds())
 		return int64(min + (max-min)*rand.Float64())
 	}
 	genNumber := rand.Float64()
 	switch {
-	case genNumber <= 0.5:
+	case genNumber < 0.5:
```

Comment:
Editorial nits: the Float64() range is [0, 1) so the comparison should be `<` and not `<=` not that it really matters. 😂